src/kontrol/foundry.py

```diff
@@ -518,32 +518,30 @@ def resolve_setup_proof_version(
         _, method = self.get_contract_and_method(test)
         effective_test_version = 0 if test_version is None else self.free_proof_version(test)
 
+        if not method.up_to_date(self.digest_file):
+            _LOGGER.info(f'Creating a new version of {test} setup proof because it was updated.')
+            return self.free_proof_version(test)
+
         if reinit:
             if user_specified_setup_version is None:
                 _LOGGER.info(
-                    f'Creating a new version of {test} because --reinit was specified and --setup-version is not specified.'
+                    f'Creating a new version of {test} setup proof because --reinit was specified and --setup-version is not specified.'
```

Comment:
Maybe we can make it more specific using something like this?
```
f'... a new version of `{test.contract.contract_name}.setUp'
```

Reply:
Ah, thanks! Do we need a separate `"setup proof"` then? But it looks fine to me either way.